components/homme/src/share/gllfvremap_mod.F90

```diff
@@ -265,22 +265,22 @@ end subroutine gfr_init
 
   subroutine gfr_init_hxx() bind(c)
 #if KOKKOS_TARGET
-    use control_mod, only: theta_hydrostatic_mode
```

Comment:
Keeping `theta_hydrostatic_mode` an int throughout the code (except for local vars for kernels) will make all these changes not needed.